builtin/logical/consul/path_token.go

```diff
@@ -90,26 +108,43 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 	}
 
 	// Create an ACLToken for Consul 1.4 and above
-	policyLink := []*api.ACLTokenPolicyLink{}
-	for _, policyName := range result.Policies {
-		policyLink = append(policyLink, &api.ACLTokenPolicyLink{
-			Name: policyName,
-		})
+	// If policies were supplied here, then overwrite the policies
+	// that were given when the role was written
+	var policyLink []*api.ACLTokenPolicyLink
+	if len(policies) > 0 {
+		policyLink = getPolicies(policies)
+	} else {
+		policyLink = getPolicies(result.Policies)
```

Comment:
Not your name, but given the size of the function, it may be a good idea to give `result` a more descriptive name.

Reply:
Refactored in https://github.com/hashicorp/vault/pull/13850/commits/dfec2cfed14973eb8d5cfe3c92f24bbc8e08c72f